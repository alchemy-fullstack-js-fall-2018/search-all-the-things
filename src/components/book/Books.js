import React, { Component, Fragment } from 'react';
import { getBooks } from '../../services/api';
import PropTypes from 'prop-types';
import Pageable from '../pageable/Pageable';
export default class Books extends Component {
    static propTypes = {
        title: PropTypes.string
    };

    state = {
        books: [],
        currentPage: 1,
        totalPages: null
    };

    updateResults = () => {
        const { title } = this.props;
        getBooks(title, this.state.currentPage).then(
            ({ books, totalPages }) => {
                this.setState({ books: books, totalPages: totalPages });
            }
        );
    };

    handlePageUpdate = page => {
        this.setState({ currentPage: page }, () => {
            this.updateResults();
        });
    };

    componentDidMount() {
        this.updateResults();
    }

    componentDidUpdate(prevProps) {
        if(prevProps.title !== this.props.title) {
            this.updateResults();
        }
    }

    render() {
        const { books, currentPage, totalPages } = this.state;
        const booksComponents = books.map(book => {
            return <Book key={book.id} book={book} />;
        });

        return (
            <Fragment>
                <Pageable
                    currentPage={currentPage}
                    totalPages={totalPages}
                    updatePage={this.handlePageUpdate}
                />
                <div>{booksComponents}</div>
            </Fragment>
        );
    }
}

const Book = ({ book }) => {
    return (
        <div>
            <p>{book.volumeInfo.title}</p>
        </div>
    );
};
