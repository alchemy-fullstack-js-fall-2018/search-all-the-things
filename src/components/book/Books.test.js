import React from 'react';
import Books from './Books';
import render from 'react-test-renderer';
jest.mock('../../services/api', () => {
    return { getBooks: () => Promise.resolve({ books: [], totalPages: 1 }) };
});


describe('books', () => {
    test('books snapshot', () => {
        expect(render
            .create(<Books title="my title" />)
            .toJSON()).toMatchSnapshot();
    });
});

