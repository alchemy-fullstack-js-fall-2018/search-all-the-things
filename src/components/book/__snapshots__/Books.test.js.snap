// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`books books snapshot 1`] = `
Array [
  <h1>
    test
  </h1>,
  <span>
    1
     out of 
  </span>,
  <div />,
]
`;
